test/diffbinder.cpp

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) 2020 Andrii Verbytskyi <andrii.verbytskyi@mpp.mpg.de> for the binder project
+//
+// All rights reserved. Use of this source code is governed by a
+// MIT license that can be found in the LICENSE file.
+#include <iostream>
+#include <fstream>
+#include <stdio.h>
+#include <vector>
+#include <string.h>
+
+int skip(std::string const & s1)
+{
+    if (s1.length()==0) return 1;
+    if (s1.length()>1) 
+    {
+        size_t comment=s1.find_first_of("//");
+        if (comment==s1.find_first_not_of(" \t")&&comment!=std::string::npos) return 2; //The comments should be ignored
+    }
+    if (s1.find("Source list file")!=std::string::npos) return 3;       //Same as the list of source files
+    if (s1.find("Modules list file")!=std::string::npos) return 4;      //Same as the list of modules
+    if (s1.find("File:")!=std::string::npos) return 5;                  //Same as the full file paths.
+    if (s1.find("pybind11::handle cl_type = cl;")!=std::string::npos) return 6;  //This could be optimized in binder
+    if (s1.find("__cxx11")!=std::string::npos) return 7;                         //This looks like implementation dependent
+    if (s1.find("assign")!=std::string::npos) return 8;                          // The assing is binded differently for some reasons
+    if (s1.find("#include")!=std::string::npos) return 9;                        // Includes could differ
+    if (s1.find("pybind11::gil_scoped_acquire gil;")!=std::string::npos) return 10; //This could be optimized in binder
+    return 0;
+}
+std::string strip(std::string const & s1)
+{
+    if (s1.length()==0) return s1;
```

Comment:
optional (feel free to ignore this): `if( s1.empty() )...`

Reply:
OK.